upgrade/v070to080/upgrade.go

```diff
@@ -2,14 +2,12 @@ package v070to080
 
 import (
 	"fmt"
-
+	"github.com/longhorn/longhorn-manager/datastore"
+	"github.com/longhorn/longhorn-manager/engineapi"
```

Comment:
Move down these 2 imports

Reply:
*
1. Create a separate commit for the upgrade path.
2. You may miss upgrading im pods*

For 1, sure. 

For 2, Can you give more details? I currently looping over all instance managers and update the labels for the ones which have `im.Spec.Image != ""`

Reply:
When we create IM pods based on the IM objects, we also set the labels for the pods. You need to update those labels for the existing IM pods during the upgrade:
https://github.com/longhorn/longhorn-manager/blob/bc77aeccd90496384e1a84e185a9eb6e7076e6f4/controller/instance_manager_controller.go#L567
https://github.com/longhorn/longhorn-manager/blob/bc77aeccd90496384e1a84e185a9eb6e7076e6f4/controller/instance_manager_controller.go#L622

Reply:
For newly created pods, the labels should be correct based on:
https://github.com/longhorn/longhorn-manager/pull/610/files#diff-9457129899e29a9925770fb1644a14f3R197-R209

For existing instance managers shouldn't the below upgrade code take care of them?
https://github.com/longhorn/longhorn-manager/pull/610/files#diff-bcdd9697a5d29e921daf6d223ba22e49R58-R68

Does the update miss any instance-managers?

Reply:
I mean the IM pods rather than the IM objects. The IM object upgrade is fine.

Reply:
Thank you Shuo and Joshua for the details! Yeah, I checked a gain and it turned out like Shuo said. The labels for the existing IM pods remain the same as before upgrading. 

I looked at the Instance Manager e  in Worker Node 1.

Here is the IM object  before upgrading:
<img width="895" alt="Screen Shot 2020-06-25 at 11 11 37 PM" src="https://user-images.githubusercontent.com/22139961/85827289-d94e6080-b73a-11ea-9d0b-efe04e2a0c13.png">


The IM object after upgrading:
<img width="1018" alt="Screen Shot 2020-06-25 at 11 11 48 PM" src="https://user-images.githubusercontent.com/22139961/85827351-f3883e80-b73a-11ea-8327-d8fd7d0aae0d.png">

The IM pod before upgrading:
<img width="1241" alt="Screen Shot 2020-06-25 at 11 17 02 PM" src="https://user-images.githubusercontent.com/22139961/85827368-03a01e00-b73b-11ea-8676-467d31315af7.png">

And the IM pod after upgrading:
<img width="731" alt="Screen Shot 2020-06-25 at 11 17 18 PM" src="https://user-images.githubusercontent.com/22139961/85827387-1155a380-b73b-11ea-9412-e94ed2531712.png">

Reply:
@joshimoo , @shuo-wu would you able to give me some thought about my question above:

-> As a result, everything is still fine because the im.name does not change during the upgrade. This leads me to a question of whether we actually need to update the labels for IM pods or not?

Reply:
The current update code will fix the existing instance manager crds and all newly created instance manager pods which will fix the customer issue longhorn/longhorn#1323. The previously existing instance manager pods will either already have correct images or they wouldn't have been able to start or failed creation to begin with.

@shuo-wu  mentioned that the labels on the existing instance manager pods are not updated, this is correct but has no real impact since we only use the pods via the instance-manager name. Considering this, I think it's okay to merge the existing code and have @PhanLe1010 create a really low priority technical debt tracking issue just in case for updating the running instance-manager pods. Most likely we will never have to actually write any code for that new issue :)

@yasker what do you think?

Reply:
We do list instance manager for various purposes, with the new mechanism those with the old labels won’t be listed. Not sure if there are going to be an issue or not. @shuo-wu ?

Reply:
Joshua is right. Longhorn won't use the image label to find the instance manager pods. Instead, it always uses the label to find the instance manager object first, then uses the object name to find the pod. 
For the live upgrade, Longhorn does use the existing **engine** manager pod to launch the new engine process. But Longhorn will directly use `engine.Status.InstanceManagerName` to find the related engine manager object. The old label of the related engine manager pod shouldn't interrupt the upgrade path.

Hence, it's OK to not update the label for instance manager pods. But just in case, we need to test the upgrade path.  (Actually, I don't think updating instance manager pods is complicated. But it will be up to Phan.)

Reply:
Thanks @shuo-wu for the explaination! Yes, I will update the instance manager pods.